.circleci/config.yml

```diff
@@ -406,42 +401,36 @@ jobs:
               bash -c "comm -13 <(sort /tmp/store-path-pre-build | grep -v '\.drv$') <(nix path-info --all | grep -v '\.drv$' | sort) | cachix push $CACHIX_NAME"
             fi
 
-      - when:
-          condition:
-            equal: [ "true", << parameters.coverage >> ]
-          steps:
-            - run:
-                name: "Persist Coverage to Workspace"
-                command: |
-                  mkdir -p coverage-workspace
-                  cp ./result*/.coverage coverage-workspace/.coverage.linux-<< parameters.py-version >>-<< parameters.tahoe-lafs-source >>
-
-            - persist_to_workspace:
-                root: "coverage-workspace"
-                paths:
-                  - "*"
+      - run:
+          name: "Persist Coverage to Workspace"
+          command: |
+            mkdir -p coverage-workspace
+            cp ./result*/.coverage coverage-workspace/.coverage.linux-<< parameters.py-version >>-<< parameters.tahoe-lafs >>
+
+      - persist_to_workspace:
+          root: "coverage-workspace"
+          paths:
+            - "*"
 
   typecheck:
     <<: *LINUX_TESTS
 
     steps:
-      - run:
-          <<: *FIX_CA_CERTS
       - run:
           <<: *SETUP_CACHIX
       - "checkout"
       - run:
           name: "Run Type Checks"
           command: |
-            nix-shell --run 'mypy src'
+            nix run .#mypy -- src
       - run:
           <<: *PUSH_TO_CACHIX
 
 
   # A job that merges all of the coverage data and generates a report.
   finish-coverage:
     docker:
-      - image: "nixos/nix:2.8.1"
+      - image: "nixos/nix:2.9.2"
```

Comment:
Seeing this version update the third time I wonder whether there's a way to DRY this up a bit - if that makes sense?

Reply:
The four approaches I know of:

* YAML anchors/references
* CircleCI orbs
* CircleCI custom executors
* write a well-factored Haskell program that spits out the correct yaml